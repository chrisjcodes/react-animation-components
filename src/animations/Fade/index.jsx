import React from 'react';
import { bool, node, number, string } from 'prop-types';
import { Transition } from 'react-transition-group';

import {
    defaultAnimationProps,
    getInlineStyles,
    getTimeoutValue,
} from 'utilities';

const Fade = props => {
    const statusStyles = {
        entered: {
            opacity: props.enterOpacity,
        },
        entering: {
            opacity: props.exitOpacity,
        },
        exited: {
            opacity: props.exitOpacity,
        },
        exiting: {
            opacity: props.exitOpacity,
        },
    };

    return (
        <Transition timeout={getTimeoutValue(props)} {...props}>
            {status => (
                <div
                    className={props.className}
                    style={{
                        ...getInlineStyles(props),
                        ...statusStyles[status],
                        transitionProperty: 'opacity',
                        willChange: 'opacity',
                    }}
                >
                    {props.children}
                </div>
            )}
        </Transition>
    );
};

Fade.propTypes = {
    appear: bool,
    children: node.isRequired,
    className: string,
    delay: number,
    duration: number,
    enterOpacity: number,
    exitOpacity: number,
    timingFn: string,
};

Fade.defaultProps = {
    ...defaultAnimationProps,
    enterOpacity: 1,
    exitOpacity: 0,
};

export default Fade;
