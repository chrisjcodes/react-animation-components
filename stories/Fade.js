import React from 'react';

import { storiesOf } from '@storybook/react';
import * as knobs from '@storybook/addon-knobs';

import { createCommonKnobs } from '../src/utilities';

import { Fade } from '../src/index';

storiesOf('Animations/Fade', module)
    .addDecorator(knobs.withKnobs)
    .add('default', () => {
        const commonKnobs = createCommonKnobs(knobs);

        const unmountOnExit = knobs.boolean('unmountOnExit', false);
        const mountOnEnter = knobs.boolean('mountOnEnter', false);

        const enterOpacity = knobs.number('enterOpacity', 1);
        const exitOpacity = knobs.number('exitOpacity', 0);

        return (
            <Fade
                in={commonKnobs.inProp}
                delay={commonKnobs.delay}
                duration={commonKnobs.duration}
                timingFn={commonKnobs.timingFn}
                unmountOnExit={unmountOnExit}
                mountOnEnter={mountOnEnter}
                enterOpacity={enterOpacity}
                exitOpacity={exitOpacity}
            >
                <h1>Example</h1>
            </Fade>
        );
    });
