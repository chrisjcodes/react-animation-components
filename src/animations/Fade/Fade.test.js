import React from 'react';
import Fade from './index.jsx';
import renderer from 'react-test-renderer';

describe('Fade', () => {
    test('renders', () => {
        const component = renderer.create(<Fade />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('can accept custom styles', () => {
        const component = renderer.create(
            <Fade style={{ background: 'black' }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionDelay with delay prop', () => {
        const component = renderer.create(<Fade delay={1000} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionDuration with duration prop', () => {
        const component = renderer.create(<Fade duration={1000} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionTimingFunction with timingFn prop', () => {
        const component = renderer.create(<Fade timingFn="ease-in-out" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
