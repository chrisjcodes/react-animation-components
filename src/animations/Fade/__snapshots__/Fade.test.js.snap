// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fade can accept custom styles 1`] = `
<div
  className={undefined}
  style={
    Object {
      "background": "black",
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
/>
`;

exports[`Fade renders 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
/>
`;

exports[`Fade sets transitionDelay with delay prop 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "1000ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
/>
`;

exports[`Fade sets transitionDuration with duration prop 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "1000ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease",
      "willChange": "opacity",
    }
  }
/>
`;

exports[`Fade sets transitionTimingFunction with timingFn prop 1`] = `
<div
  className={undefined}
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease-in-out",
      "willChange": "opacity",
    }
  }
/>
`;
